doc/api/n-api.md

```diff
@@ -4672,7 +4672,7 @@ napi_status napi_async_init(napi_env env,
 ```
 
 * `[in] env`: The environment that the API is invoked under.
-* `[in] async_resource`: An optional object associated with the async work
+* `[in] async_resource`: Object associated with the async work
   that will be passed to possible `async_hooks` [`init` hooks][].
```

Comment:
```suggestion
  that will be passed to possible `async_hooks` [`init` hooks][].
  In order to retain ABI compatibility with previous versions,
  passing `NULL` for `async_resource` will not result in an error, however,
  this will result incorrect operation of async hooks for the
  napi_async_context created. Potential issues include
  loss of async context when using the AsyncLocalStorage API.
```